{
  "name": "assertion-error",
  "version": "2.0.1",
  "description": "Error constructor for test and validation frameworks that implements standardized AssertionError specification.",
  "author": {
    "name": "Jake Luer",
    "email": "jake@qualiancy.com",
    "url": "http://qualiancy.com"
  },
  "license": "MIT",
  "types": "./index.d.ts",
  "keywords": [
    "test",
    "assertion",
    "assertion-error"
  ],
  "repository": {
    "type": "git",
    "url": "git@github.com:chaijs/assertion-error.git"
  },
  "engines": {
    "node": ">=12"
  },
  "files": [
    "index.d.ts"
  ],
  "type": "module",
  "module": "index.js",
  "main": "index.js",
  "scripts": {
    "build": "deno bundle mod.ts > index.js",
    "pretest": "rm -rf coverage/",
    "test": "deno test --coverage=coverage",
    "posttest": "deno coverage coverage --lcov > coverage/lcov.info && lcov --summary coverage/lcov.info"
  },
  "readme": "<p align=center>\n  AssertionError and AssertionResult classes.\n</p>\n\n<p align=center>\n  <a href=\"https://github.com/chaijs/assertion-error/actions\">\n    <img\n      alt=\"build:?\"\n      src=\"https://github.com/chaijs/assertion-error/actions/workflows/nodejs.yml/badge.svg\"\n    />\n  </a><a href=\"https://www.npmjs.com/package/assertion-error\">\n    <img\n      alt=\"downloads:?\"\n      src=\"https://img.shields.io/npm/dm/assertion-error.svg\"\n    />\n  </a><a href=\"\">\n    <img\n      alt=\"devDependencies:none\"\n      src=\"https://img.shields.io/badge/dependencies-none-brightgreen\"\n    />\n  </a>\n</p>\n\n## What is AssertionError?\n\nAssertion Error is a module that contains two classes: `AssertionError`, which\nis an instance of an `Error`, and `AssertionResult` which is not an instance of\nError.\n\nThese can be useful for returning from a function - if the function \"succeeds\"\nreturn an `AssertionResult` and if the function fails return (or throw) an\n`AssertionError`.\n\nBoth `AssertionError` and `AssertionResult` implement the `Result` interface:\n\n```typescript\ninterface Result {\n  name: \"AssertionError\" | \"AssertionResult\";\n  ok: boolean;\n  toJSON(...args: unknown[]): Record<string, unknown>;\n}\n```\n\nSo if a function returns `AssertionResult | AssertionError` it is easy to check\n_which_ one is returned by checking either `.name` or `.ok`, or check\n`instanceof Error`.\n\n## Installation\n\n### Node.js\n\n`assertion-error` is available on [npm](http://npmjs.org).\n\n```\n$ npm install --save assertion-error\n```\n\n### Deno\n\n`assertion_error` is available on\n[Deno.land](https://deno.land/x/assertion_error)\n\n```typescript\nimport {\n  AssertionError,\n  AssertionResult,\n} from \"https://deno.land/x/assertion_error@2.0.0/mod.ts\";\n```\n",
  "readmeFilename": "README.md",
  "bugs": {
    "url": "https://github.com/chaijs/assertion-error/issues"
  },
  "_id": "assertion-error@2.0.1",
  "dist": {
    "shasum": "78e473ebde769376bcc672e6f22a8d375db57235"
  },
  "_from": "assertion-error@^2.0.1",
  "_resolved": "https://registry.npmjs.org/assertion-error/-/assertion-error-2.0.1.tgz"
}
